### DISCUSSION TITLE
Starting Problems with XIAO ESP32S3

### DISCUSSION THREAD

**Discussion by mediaessenz** [pschatzmann/arduino-audio-tools#1644] [General]
I have problems to get the compilation of any example, even the simplest working.
I tried using arduino ide (2.3.2) and vsc with the platformio plugin.

Maybe someone can help me to get at least the streams_generator_serial example in the Arduino IDE working?

Under board manager I have installed the esp32 from expressif in version 3.0.4.
Under Libraries I have installed "audio-tools" from Phil Schatzmann in version 0.9.9

After this I choose my board "XIAO_ESP32S3" and the usb port "/dev/cu.usbmodem1201" on my mac.

If I try to compile the example code "streams-generator-serial.ino":
```c++
/**
 * @file streams-generator-serial.ino
 * @author Phil Schatzmann
 * @brief see https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-stream/streams-generator-serial/README.md 
 * @copyright GPLv3
 **/
 
#include "AudioTools.h"


AudioInfo audio_info(44100, 2, 16);
SineWaveGenerator<int16_t> sineWave(32000);                // subclass of SoundGenerator with max amplitude of 32000
GeneratedSoundStream<int16_t> sound(sineWave);             // Stream generated from sine wave
CsvOutput<int16_t> out(Serial); 
StreamCopy copier(out, sound);                             // copies sound to out

// Arduino Setup
void setup(void) {  
  // Open Serial 
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Warning);

  // Define CSV Output
  out.begin(audio_info);

  // Setup sine wave
  sineWave.begin(audio_info, N_B4);
  Serial.println("started...");
}

// Arduino loop - copy sound to out 
void loop() {
  copier.copy();
}
```

I get this error:
```
In file included from /Users/alex/Documents/Arduino/libraries/arduino-audio-tools/src/AudioAnalog/AnalogAudioStream.h:7,
                 from /Users/alex/Documents/Arduino/libraries/arduino-audio-tools/src/AudioTools.h:80,
                 from /Users/alex/Documents/Arduino/streams_generator_serial/streams_generator_serial.ino:8:
/Users/alex/Documents/Arduino/libraries/arduino-audio-tools/src/AudioAnalog/AnalogDriverESP32V1.h: In member function 'bool audio_tools::AnalogDriverESP32V1::setup_rx()':
/Users/alex/Documents/Arduino/libraries/arduino-audio-tools/src/AudioAnalog/AnalogDriverESP32V1.h:525:20: error: 'struct adc_continuous_handle_cfg_t' has no member named 'flags'
  525 |         adc_config.flags.flush_pool  = true;
      |                    ^~~~~

exit status 1

Compilation error: exit status 1
```
I already spent hours to find the reason of this problem 😢

**Comment by pschatzmann** [comment]
adc_config.flags is only available starting from IDF 5.2, but Arduino is still on 5.1.4

Delete the arduino-audio-tools in the libraries directory and try to reinstall it again.
This should install the corrected library

**Comment by mediaessenz** [comment]
If I deinstall and install the audio-tools again, I get this error:
```
Library 'audio-tools:0.9.9' could not be installed. No valid dependencies solution found: dependency 'I2S' is not available
```
I had this error before again, that's why I install it over the console like so:
```
cd  ~/Documents/Arduino/libraries
git clone https://github.com/pschatzmann/arduino-audio-tools.git
```
I did now install it again this way, and the adc_config.flags error is gone. 

What stands "IDF" for?

**Reply by pschatzmann** [reply]
I was not really prepared to support the installation via the Arduino Library manager and somebody added my libraries w/o asking me.

Try again. 

If I am getting too many issues because of this, I will have them removed again

IDF is the Espressif IoT Development Framework...

**Reply by pschatzmann** [reply]
I requested the removal from the Arduino Library Manager.

Please install the library as described in the README

**Reply by mediaessenz** [reply]
Great, I'm sure this will help a lot of beginners.

Thank you for all.
